code/messaging/src/androidTest/java/com/adobe/marketing/mobile/messaging/internal/MessagingPublicAPITests.java

```diff
@@ -400,6 +401,75 @@ public void testRefreshInAppMessages() throws InterruptedException {
         assertEquals(expectedEdgePersonalizationEventData.trim(), edgePersonalizationRequestEvents.get(0).getEventData().toString());
     }
 
+    // --------------------------------------------------------------------------------------------
+    // Tests for Messaging.updateFeedsForSurfacePaths API
+    // --------------------------------------------------------------------------------------------
+    @Test
+    public void testUpdateFeedsForSurfacePaths() throws InterruptedException {
+        // setup
+        List<String> surfacePaths = new ArrayList<>();
+        surfacePaths.add("promos/feed1");
+        surfacePaths.add("promos/feed2");
+        final String expectedMessagingEventData = "{updatefeeds=true, surfaces=[promos/feed1, promos/feed2]}";
+        final String expectedEdgePersonalizationEventData = "{xdm={eventType=personalization.request}, query={personalization={surfaces=[mobileapp://com.adobe.marketing.mobile.messaging.test/promos/feed1, mobileapp://com.adobe.marketing.mobile.messaging.test/promos/feed2]}}}";
+        // test
+        Messaging.updateFeedsForSurfacePaths(surfacePaths);
+        TestHelper.sleep(500);
+
+        // verify messaging request content event
+        final List<Event> messagingRequestEvents = getDispatchedEventsWith(MessagingTestConstants.EventType.MESSAGING,
+                EventSource.REQUEST_CONTENT);
+        assertEquals(1, messagingRequestEvents.size());
+        assertEquals(expectedMessagingEventData, messagingRequestEvents.get(0).getEventData().toString());
```

Comment:
This is error prone and difficult to update/ maintain. Can we please not convert to string to compare in the tests here?